client/http_sender.go

```diff
@@ -109,7 +109,7 @@ func (s *httpSender) Send(_ context.Context, call Call) {
 					// Retry on service unavailable and request timeout.
 					// TODO(spencer): consider respecting the Retry-After header for
 					// backoff / retry duration.
-					return retry.Continue, nil
+					return retry.Continue, err
```

Comment:
The only thing an error returned with retry.Continue is used for is logging; we should remove logging that would be redundant as we change nil to err in cases like this (and maybe change the error returned to have the message that we would have logged)